File: evm/python/examples/read_role.py
"""Read a deployed role using native Web3.py contract calls."""

import os

from web3 import Web3

from swig_evm.contracts import get_contract
from swig_evm.roles import decode_role

web3 = Web3(Web3.HTTPProvider(os.environ["RPC_URL"]))
config = get_contract(web3, "SwigConfig", os.environ["ACCOUNT_ADDRESS"])
print(decode_role(config.functions.getRole(int(os.environ.get("ROLE_ID", "0"))).call()))

File: evm/rust/examples/read_role.rs
use alloy::{primitives::Address, providers::ProviderBuilder};
use std::{env, error::Error};
use swig_evm::{contracts::config::SwigConfig, roles::Role};

#[tokio::main]
async fn main() -> Result<(), Box<dyn Error>> {
    let provider = ProviderBuilder::new().connect_http(env::var("RPC_URL")?.parse()?);
    let address: Address = env::var("ACCOUNT_ADDRESS")?.parse()?;
    let role_id = env::var("ROLE_ID").unwrap_or_else(|_| "0".into()).parse()?;
    let config = SwigConfig::new(address, provider);
    println!(
        "{:?}",
        Role::decode(&config.getRole(role_id).call().await?)?
    );
    Ok(())
}
